src/RenderMethods/RenderFieldTemplate.tsx

```diff
@@ -0,0 +1,135 @@
+  const { schema: vernaSchema, setSchema, uiSchema, setUiSchema } = useVerna();
+  const canAddField = id.split('_').length === 3;
+  const canAddSection = id.split('_').length === 2;
+  const isRoot = id === 'root';
+  const ownProperties = Object.keys(schema.properties || {}).length === 0;
+  const currentSection = getCurrentSection(id);
+  const currentField = getCurrentField(id);
+
+  function addItemToSchema(newKey: string) {
+    const newSchema = { ...vernaSchema };
+    if (newSchema.properties) {
+      const section = newSchema.properties[currentSection] as JSONSchema7;
+      if (section.properties) {
+        section.properties[newKey] = stringDefinition(newKey);
+      }
+    }
+    setSchema(newSchema);
+  }
+
+  function addItemToUiSchemaOrder(newKey: string) {
+    const newUiSchema = { ...uiSchema };
+    if (currentSection in newUiSchema && 'ui:order' in newUiSchema[currentSection]) {
+      const currentSectionIndex = uiSchema[currentSection]['ui:order'].findIndex(
+        (key: string) => key === currentField,
+      );
+      newUiSchema[currentSection]['ui:order'].splice(currentSectionIndex + 1, 0, newKey);
+    } else {
+      newUiSchema[currentSection] = {};
+      newUiSchema[currentSection]['ui:order'] = [newKey];
+    }
+    setUiSchema(newUiSchema);
+  }
+
+  function addItem() {
+    const newKey = makeid(10);
+    addItemToSchema(newKey);
+    addItemToUiSchemaOrder(newKey);
+  }
+
+  function addSection() {
+    const newSchema = { ...vernaSchema };
+    const newKey = makeid(10);
+    if (newSchema.properties) {
+      newSchema.properties[newKey] = sectionDefinition(newKey);
+    }
+    setSchema(newSchema);
+    const currentSection = getCurrentSection(id);
+    const newUiSchema = { ...uiSchema };
+    if (newUiSchema['ui:order']) {
+      const currentSectionIndex =
+        newUiSchema['ui:order'].findIndex((key) => key === currentSection) || 0;
+      newUiSchema['ui:order'].splice(currentSectionIndex + 1, 0, newKey);
+      setUiSchema(newUiSchema);
+    }
+  }
+
+  function removeItem() {
+    const newSchema = { ...vernaSchema };
+    if (newSchema.properties) {
+      const sectionProperties = newSchema.properties[currentSection] as JSONSchema7;
+      if (sectionProperties.properties) {
+        delete sectionProperties.properties[currentField];
+      }
+    }
+    setSchema(newSchema);
+    if (uiSchema[currentSection]['ui:order']) {
+      const newUiSchema = { ...uiSchema };
+      newUiSchema[currentSection]['ui:order'] = newUiSchema[currentSection]['ui:order'].filter(
+        (key: string) => key !== currentField,
+      );
+      setUiSchema(newUiSchema);
+    }
+  }
+
+  function removeSection() {
+    const newSchema = { ...vernaSchema };
+    const newUiSchema = { ...uiSchema };
+    if (newSchema.properties) {
+      delete newSchema.properties[currentSection];
+    }
+    setSchema(newSchema);
+    delete newUiSchema[currentSection];
```

Comment:
Just for your information, you could write something like that to quickly get `newUiSchema` without `currentSection`
```suggestion
    const {[currentSection]:_, ...newUiSchema} = uiSchema;
```

Reply:
https://c.tenor.com/B7vSc-79QXAAAAAd/mindblow-mind-explosion.gif
Didn't knew this way of doing it ! Thanks !